Plutarch/Api/V1/Value.hs

```diff
@@ -270,7 +270,7 @@ psingleton = phoistAcyclic $
   plam $ \symbol token amount ->
     pif
       (amount #== 0)
-      mempty
+      (ptraceError "psingleton: Given a zero argument, expecting nonzero.")
```

Comment:
This is a huge breaking change. I suggest either changing the type to use GDP or make a new function.

Reply:
I would argue the original implementation is a bug. The name 'singleton' has pretty strong implications of behaviour: I certainly don't expect any _other_ function with such a name to potentially return an empty structure sometimes.

Reply:
At _minimum_, the documentation should be yelling this from the rooftops.

Reply:
When is the current behaviour problematic in practice? I think it's precisely what you want.

Reply:
If I give 0 as an argument, what I am getting is not a 'singleton', since it's empty. I would argue that this is misleading _at best_. Thus, either the name is wrong, or the semantics are wrong.

Reply:
But this is a map from an index to a monoid. When do you care whether the entry is there or not when you want the value to be zero?

Reply:
Yeah, upon reading the comment regarding currencies in `plutus-ledger-api`, that makes sense. Gonna withdraw this PR.